cli/connhelper/connhelper.go

```diff
@@ -91,3 +99,15 @@ func disablePseudoTerminalAllocation(sshFlags []string) []string {
 	}
 	return append(sshFlags, "-T")
 }
+
+// dockerSSHRemoteBinary returns the binary to use when executing Docker
+// commands over SSH. It defaults to "docker" if the DOCKER_SSH_REMOTE_BINARY
+// environment variable is not set.
+func dockerSSHRemoteBinary() string {
+	value := os.Getenv(DockerSSHRemoteBinaryEnv)
```

Comment:
It would probably be better if this was a per-context setting rather than a global env-variable.
However, it's a niche use-case, so I guess it's _okay_ as an escape hatch and not really a blocker.

@thaJeztah @laurazard WDYT?